vars/getNewVersion.groovy

```diff
@@ -9,14 +9,11 @@ def call(body) {
 
     // until we fix up pushing tags to remote repos with correct secrets lets default to use the short commit sha as the version
     echo 'NOTE: until we support pushing tags to remote repos with correct secrets lets default to use the short commit sha as the version'
-    def version 
     try {
-        version = sh(script: 'git rev-parse --short HEAD', returnStdout: true).toString().trim()
-    } catch (err){
+        def version = 'v' + sh(script: 'git rev-parse --short HEAD', returnStdout: true).toString().trim()
```

Comment:
[`error`][error] aborts the build in a graceful manner so, I think there nothing wrong with the old implementation which catches any error while executing git command and  gracefully terminates the build with a message.

[error]: https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#code-error-code-error-signal

Reply:
@sthaha conceptually equivalent, but version might be undefined if catch returns early, I made the scope of the variable shorter and explicit inside the try block.

Reply:
@sthaha  I'm just reducing the scope of the version variable to the try block where it actually lives and removing a mutation.

Reply:
ok .. I don't feel strongly about the new or the old. To me both are equally good. The old one had a return at the end of the method which (to me) was simpler